scripts/generate-shaders.js

```diff
@@ -82,18 +81,21 @@ ${fragmentPrelude}
                     "{name} = unpack_mix_{a_type}(a_{name}, a_{name}_t);"
                 ]
             });
-    }
 
-    function fragmentSource() {
-        const source = fs.readFileSync(path.join(inputPath, shaderName + '.fragment.glsl'), 'utf8');
-        return applyPragmas(source, {
+    const fragmentSource = applyPragmas(fs.readFileSync(path.join(inputPath, shaderName + '.fragment.glsl'), 'utf8'), {
                 define: [
                     "varying {precision} {type} {name};"
                 ],
                 initialize: [
                 ]
             });
-    }
+
+    const shasum = crypto.createHash('sha1');
```

Comment:
We are using `std::hash` now to hash both shaders, convert the result to a hexadecimal string, and concatenate them. This results in 64 bits of entropy on 32 bit systems, and 128 bits of entropy on 64 bit systems. @jfirebaugh do you think `std::hash` is enough to protect us from identical strings, or should we use a different hash function?